### PR TITLE
feat: prototype network request interception: scaffold protocol

### PR DESCRIPTION
**Pull request by thiagowfx**
Bug: #644
Doc: http://go/webdriver-bidi:network-request-interception-proposal

### CODE REVIEW

**Comment by thiagowfx** [comment]
All data structures have been added

**Review by jrandolf — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -221,7 +221,7 @@
     "source-map-support": "0.5.21",
     "terser": "5.18.2",
     "tslib": "2.6.0",
-    "typescript": "5.1.5",
+    "typescript": "5.1",
```
Comment (Lightning00Blade) [inline package.json:224]: Why this change?

Comment (thiagowfx) [reply package.json:224]: ESLint (via pre-commit) had a warning along the lines that it's only supported up to 5.1 (ESTree).

**Review by Lightning00Blade — APPROVED** [review]
